File: Observer/Observer.py
from abc import ABC, abstractmethod

class Observer(ABC):
    @abstractmethod
    def notify(self):
        pass

class FirstObserver(Observer):
    def notify(self):
        print("FirstObserver notified")

class SecondObserver(Observer):
    def notify(self):
        print("SecondObserver notified")

class Subject:
    observers: list
    def __init__(self):
        self.observers = []
    def add_observer(self, observer: Observer):
        self.observers.append(observer)
    def remove_observer(self, obs: Observer):
        for index, observer in enumerate(self.observers):
            if observer == obs:
                self.observers = self.observers[:index] + self.observers[index + 1:]
                break
    def notify_observers(self):
        for observer in self.observers:
            observer.notify()
    
if __name__ == "__main__":
    subject = Subject()
    obs1 = FirstObserver()
    obs2 = SecondObserver()
    subject.add_observer(obs1)
    subject.notify_observers()
    print("#################")

    subject.add_observer(obs2)
    subject.notify_observers()
    print("#################")

    subject.remove_observer(obs1)
    subject.notify_observers()

File: Observer/Observer.java
import java.util.HashSet;
import java.util.Set;

interface Observer {
    void notified();
} 

class ConcreteObserver implements Observer{
    String name;
    ConcreteObserver(String name)
    {
        this.name = name;
    }
    public void notified() {
        System.out.println(name + " was notified.");
    }
}

class Subject {
    Set<Observer> observers = new HashSet<>();
    public void addObserver(Observer obs)
    {
        observers.add(obs);
    }
    public void removeObserver(Observer obs)
    {
        observers.remove(obs);
    }
    public void notifyObservers()
    {
        for(Observer obs: observers)
        {
            obs.notified();
        }
    }
}

class Main {
    public static void main(String[] args) 
    {
        Subject subject = new Subject();
        Observer first = new ConcreteObserver("First");
        Observer second = new ConcreteObserver("Second");
        subject.addObserver(first);
        subject.addObserver(second);
        subject.notifyObservers();

        subject.removeObserver(second);
        subject.notifyObservers();
    }
}
